lib/net.js

```diff
@@ -1535,6 +1535,7 @@ function onconnection(err, clientHandle) {
   }
 
   if (self.maxConnections && self._connections >= self.maxConnections) {
+    self.emit('maxConnections');
```

Comment:
@mcollina @addaleax @BridgeAR ... would be good to get some more opinions on this... do we want to communicate this case this way or using an error instead?

Reply:
I think this is totally fine as is.